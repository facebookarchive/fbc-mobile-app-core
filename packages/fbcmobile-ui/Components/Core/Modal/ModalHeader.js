/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import type {ViewStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';

import * as React from 'react';
import Text from '@fbcmobile/ui/Components/Core/Text';
import {Colors} from '@fbcmobile/ui/Theme';
import {Icon} from 'react-native-material-ui';
import {StyleSheet, TouchableOpacity, View} from 'react-native';

type Props = {
  +title: React.Node,
  +onClosePressed?: () => void,
  +showCloseButton?: boolean,
  +style?: ViewStyleProp,
};

const ModalHeader = ({
  title,
  onClosePressed,
  style,
  showCloseButton = true,
}: Props) => {
  return (
    <View style={[styles.root, style]}>
      {showCloseButton ? (
        <TouchableOpacity style={styles.closeButton} onPress={onClosePressed}>
          <Icon
            iconSet="MaterialCommunityIcons"
            name="window-close"
            color={Colors.Gray60}
            size={28}
          />
        </TouchableOpacity>
      ) : null}
      <View style={styles.textContainer}>
        <Text
          variant="h3"
          weight="bold"
          style={showCloseButton ? styles.text : styles.textCentered}>
          {title}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 34,
  },
  closeButton: {
    paddingLeft: 16,
  },
  textContainer: {
    flex: 1,
    paddingHorizontal: 10,
    alignItems: 'flex-start',
    justifyContent: 'center',
  },
  text: {
    width: '100%',
    textAlign: 'left',
  },
  textCentered: {
    width: '100%',
    textAlign: 'center',
  },
});

export default ModalHeader;
