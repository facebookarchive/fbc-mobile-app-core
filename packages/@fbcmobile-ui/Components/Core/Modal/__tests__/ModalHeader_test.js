/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import ModalHeader from '../ModalHeader';
import React from 'react';
import renderer from 'react-test-renderer';

describe('ModalHeader Tests', () => {
  test('ModalHeader renders correctly', () => {
    const tree = renderer.create(<ModalHeader title={'title'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('ModalHeader renders correctly without show close button', () => {
    const tree = renderer
      .create(<ModalHeader title={'title'} showCloseButton={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
