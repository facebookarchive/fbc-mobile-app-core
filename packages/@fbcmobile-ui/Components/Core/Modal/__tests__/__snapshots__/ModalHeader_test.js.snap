// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader Tests ModalHeader renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 34,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingLeft": 16,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#8F949D",
            "fontSize": 28,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ó°–­
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "color": "#484848",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
            },
            null,
            Object {
              "fontFamily": "Roboto-Bold",
            },
            Object {
              "textAlign": "left",
            },
          ],
          Object {
            "textAlign": "left",
            "width": "100%",
          },
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;

exports[`ModalHeader Tests ModalHeader renders correctly without show close button 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 34,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "color": "#484848",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
            },
            null,
            Object {
              "fontFamily": "Roboto-Bold",
            },
            Object {
              "textAlign": "left",
            },
          ],
          Object {
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;
